### DISCUSSION TITLE
ASYNCIFY_ONLY  problem with format

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#18100] [Q&A]
I am stuck with a stupid thing:  I get an error when setting ASYNCIFY_ONLY with function that get more then one parameter and I didn't find the right string format.
when setting` -s ASYNCIFY_ONLY=["CTest::test111(int,int)"]` I get the warning:` emcc:WARNING: emcc: ASYNCIFY list contains an item without balanced parentheses ("(", ")"):`
trying setting the following also failed:
```
-s ASYNCIFY_ONLY="[CTest::test111(int,int)]"
-s ASYNCIFY_ONLY='["CTest::test111(int, int)"]'
-s ASYNCIFY_ONLY=[CTest::test111(int, int)]
```
 
what am I missing ?

Emcc: 3.1.9 

command line:
C:\Projects\simple_opengl-master\TheMainProject>C:\emsdk\upstream\emscripten\\emcc.bat Emscripten/Debug/Source.o -lCore.Emscripten -s PROXY_TO_PTHREAD  -s OFFSCREENCANVAS_SUPPORT   -s OFFSCREEN_FRAMEBUFFER -profiling  -g   -s EXPORTED_RUNTIME_METHODS=ccall,cwrap -s EXPORTED_FUNCTIONS=_main,_NativeTestThreads  --preload-file com.skyline.terraexplorer  -s OFFSCREEN_FRAMEBUFFER=1 -s OFFSCREENCANVAS_SUPPORT=1  -s WASM_BIGINT  -fexceptions -s WASM_BIGINT  -s FETCH=1    --bind -fexceptions   -s USE_PTHREADS    -s ALLOW_MEMORY_GROWTH=1    -lGL -s FETCH  -s ASYNCIFY  -s ASYNCIFY_ONLY="['CTest::test111(int, int)]" -o C:/Projects/simple_opengl-master/Emscripten/Debug/TheMainProject.js -LC:/Projects/simple_opengl-master/Emscripten/Debug/

Error:
emcc:WARNING: emcc: ASYNCIFY list contains an item without balanced parentheses ("(", ")"):
emcc:WARNING:    CTest::test111(int
emcc:WARNING: This may indicate improper escaping that led to splitting inside your names.
emcc:WARNING: Try using a response file. e.g: -sASYNCIFY_ONLY=@funcs.txt. The format is a simple
emcc:WARNING: text file, one line per function.

**Comment by sbc100** [comment]
I recommend using the for @filename syntax.  e.g . `-sASYNCIFY_ONLY=@async_only.txt`.   Then you would list your symbols, one per line, in the file called `async_only.txt`, avoiding the need to escape stuff on that command line.

**Comment by aharondavid** [comment]
Ok, I will.
Thanks for your advise.
